platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/StateSignatureTransaction.java

```diff
@@ -142,50 +74,43 @@ public int getSize() {
      */
     @Override
     public void serialize(final SerializableDataOutputStream out) throws IOException {
-        out.writeByteArray(stateSignature.getSignatureBytes());
+        final StateSignaturePayload stateSignaturePayload = payload.as();
 
-        // state hash will only be null for objects originally serialized with version ORIGINAL
-        if (stateHash == null) {
-            out.writeByteArray(new byte[0]);
-        } else {
-            out.writeByteArray(stateHash.getValue());
-        }
-        out.writeLong(round);
-        out.writeSerializable(epochHash, false);
+        out.writeInt((int) stateSignaturePayload.signature().length());
```

Comment:
`writeByteArray` requires access to the underlying `byte[]` which `Bytes` does not provide, it only allows you to create a copy. for deserialization it works because I first create a `byte[]` then wrap it in `Bytes`.

All of this is quite messy because I wanted to switch to using protobufs but not migrate the serilization and hashing yet. once we migrate, we will simply delete this class, so all of this is just temporary.